Account 2FA enforcement article (#421)
dnsimple/dnsimple-support: content/articles/two-factor-authentication-enforcement.markdown

---

> title: Enforce Two-Factor Authentication for a shared Account

Reviewer: I suggest

> Enforce Two-Factor Authentication for all members of an account

The term _shared_ is ambiguous as we use it for sharing a domain. It's not really a shared account, it's just an account with multiple members.

> excerpt: How to protect your shared DNSimple account using Two-Factor authentication enforcement.

Reviewer: ```suggestion
excerpt: How to ensure an account is secured by enforcing all the members of an account to use two-factor authentication.
```

categories:
- Account
---

# Enforce Two-Factor Authentication for a shared Account

<info>
This article describes a feature in Public Beta.
</info>

### Table of Contents {#toc}

* TOC
{:toc}

---

> DNSimple supports [Two-Factor authentication](/articles/two-factor-authentication/). This feature enforces that all users in our account must have Two-Factor authentication enabled.

Reviewer: ```suggestion
[Two-Factor authentication](/articles/two-factor-authentication/) provides an extra level of security to protect your user credentials. You can enforce account members to use two-factor authentication by enabling the feature for the account.
```

I replaced the concept of _your_ account as it seems you own an account. Whereas, in fact, the account is not owned by anyone, but members belong to an account.

> When you have added other users to your account with it is important to ensure the security of each account member. This is why this feature is available to all plans that have access to the teams feature.

Reviewer: This sentence is quite confusing. I can't understand what it means. Also, please let's not refer as the Team feature.

## Enabling Two-Factor authentication enforcement {#enable}

You can enable Two-Factor authentication enforcement from your account members page.

> You can only enable Two-Factor authentication enforcement for your account, if all users have Two-Factor authentication enabled. You either need to bug them to enable it or remove them from the team to enable this feature.

Reviewer: Removing _your_ account. Users have multiple accounts, and they don't are the unique owners.

```suggestion
You can only enable Two-Factor authentication enforcement an account only if all users have Two-Factor authentication enabled. You either need to bug them to enable it or remove them from the team to enable this feature.
```

I think the following line may need some rewording:

> You either need to bug them to enable it or remove them from the team to enable this feature.

> You can only enable Two-Factor authentication enforcement an account if all users have Two-Factor authentication enabled. You either need to bug them to enable it or remove them from the team to enable this feature, then re-invite them, they will be asked to enable Two-Factor authentication after they sign up so they can join your account.

Reviewer: > You either need to bug them

This sentence here seems quite colloquial to me, not exactly something I would read in a documentation.

    ![Two-Factor authentication enforcement](/files/account-2fa-enforcement.png)

1.  If there are users in your account that does not have Two-Factor authentication enabled you will see a list of users that are not compliant, you can [remove the users](/articles/account-users/#removing-members-from-an-account) from the account in the section above.
</div>

## Disabling Two-Factor authentication enforcement {#disable}

<div class="section-steps" markdown="1">
##### To disable Two-Factor authentication

1.  Log in to DNSimple with your user credentials.
1.  Go to your account settings by selecting the <label>Settings</label> or <label>Account Settings</label> link from the navigation.

    ![Settings menu](/files/account-settings-access.jpg)

1.  Select <label>Team</label> from the menu on the left.
1.  Scroll down until you see the <label>Enforce two-factor authentication</label> section. There is a link to <label>Disable</label> the feature.

    ![Disable Two-Factor authentication enforcement](/files/account-disable-2fa-enforcement.png)
</div>

## Disabling Two-Factor authentication while in an enforced team

> When you are part of an Two-Factor enforcing team, you can still [disable Two-Factor authentication](http://localhost:3000/articles/two-factor-authentication/#disable) on your user.

Reviewer: Link to localhost. Also, a "When you are part of an Two-Factor enforcing team" seems more a special punishing squad :trollface:

> When you are part of an Two-Factor enforcing team, you can still [disable Two-Factor authentication](/articles/two-factor-authentication/#disable) on your user.

Reviewer: > "When you are part of an Two-Factor enforcing team" seems more a special punishing squad

It reads quite badly to me. It seems you are referring to an action team.

Reviewer 2: I disbanded the action squad.

<warning>
If you are disabling Two-Factor authentication you will be removed from all accounts that enforce Two-Factor authentication.
</warning>

## Inviting new people into an 2FA enforcing team

There are no restrictions on inviting new people to your account when you have Two-Factor authentication enforcement enabled. If the user already has a DNSimple account they can't join the team when they don't have Two-Factor authentication enabled.

> When you invite users to join your account that do not have an DNSimple account, they will be asked to enable Two-Factor authentication after they sign up so they can join your account.

Reviewer: To me this one should be the first sentence. The first sentence is confusing. I also believe you used the term account for user, but they are not the same thing.
